File: programers/1-체육복.js
//테스트 13, 18 실패
function solution(n, lost, reserve) {
  let realLost = lost.filter((element) => !reserve.includes(element));
  let realReserve = reserve.filter((element) => !lost.includes(element));

  let fail = realLost.filter((lost) => {
    const lend = realReserve.find((reverse) => Math.abs(reverse - lost) == 1);

    if (!lend) return lost;
    realReserve = realReserve.filter((reverse) => reverse !== lend);
  }).length;
  return n - fail;
}

File: programers/1-체육복.py
def solution(n, lost, reserve):
    cnt = 0
    real_reserve = list(set(reserve) - set(lost))
    real_lost = list(set(lost)- set(reserve))
    for i in real_lost:
        
        if (i - 1) in real_reserve:
            real_reserve.pop(real_reserve.index(i - 1))
        elif (i + 1) in real_reserve:
            real_reserve.pop(real_reserve.index(i + 1))
        else:
            cnt += 1
    answer = n - cnt
    return answer